src/app/components/category-products/category-products.component.ts

```diff
@@ -1,30 +1,138 @@
-import { combineLatest, Observable } from 'rxjs';
-import { map, shareReplay, switchMap } from 'rxjs/operators';
+import { BehaviorSubject, Observable, combineLatest, of, pipe } from 'rxjs';
+import {
+  map,
+  shareReplay,
+  startWith,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs/operators';
 import { CategoryModel } from '../../models/category.model';
+import { ProductQueryModel } from '../../query-models/product.query-model';
 import { ProductModel } from '../../models/product.model';
 import { ProductsService } from '../../services/products.service';
 import { CategoriesService } from '../../services/categories.service';
-import { ProductQueryModel } from 'src/app/query-models/product.query-model';
 
 @Component({
   selector: 'app-category-products',
   styleUrls: ['./category-products.component.scss'],
   templateUrl: './category-products.component.html',
   encapsulation: ViewEncapsulation.Emulated,
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CategoryProductsComponent {
-  readonly categories$: Observable<CategoryModel[]> = this._categoriesService.getAllCategories();
-  readonly category$: Observable<CategoryModel> = this._activatedRoute.params.pipe(switchMap(data => this._categoriesService.getOneCategory(data["categoryId"])), shareReplay(1));
-  readonly products$: Observable<ProductQueryModel[]> = combineLatest([
+
+  readonly sort: FormGroup = new FormGroup({
+    sortValue: new FormControl('', Validators.required),
+  });
+
+  readonly sortValue$: Observable<string> = this.sort.valueChanges.pipe(
+    map((form) => form.sortValue),
+    startWith('')
+  );
+
+  private _pageSizeSubject: BehaviorSubject<number> =
+    new BehaviorSubject<number>(5);
+  public pageSize$: Observable<number> = this._pageSizeSubject.asObservable();
+
+  private _pageNumberSubject: BehaviorSubject<number> =
+    new BehaviorSubject<number>(1);
+  public pageNumber$: Observable<number> =
+    this._pageNumberSubject.asObservable();
+
+  readonly categories$: Observable<CategoryModel[]> =
+    this._categoriesService.getAllCategories();
+  readonly category$: Observable<CategoryModel> =
+    this._activatedRoute.params.pipe(
+      switchMap((data) =>
+        this._categoriesService.getOneCategory(data['categoryId'])
+      ),
+      shareReplay(1)
+    );
+  readonly sortedProducts$: Observable<ProductQueryModel[]> = combineLatest([
     this._productsService.getAllProducts(),
-    this.category$
-  ]).pipe(map(([products, category]) => this._mapToProductQueryModels(products).filter((product) => product.categoryId === category.id).sort((a, b) => +b.featureValue - +a.featureValue)))
+    this.category$,
+    this.sortValue$,
+  ]).pipe(
+    map(([products, category, sortValue]) =>
+      this._mapToProductQueryModels(products)
+        .filter((product) => product.categoryId === category.id)
+        .sort((a, b) => {
+          if (sortValue === 'price-') {
```

Comment:
you can simplify this code still. Nice you use the price / ratingValue as id - this is the way to go. you are only missing to extract information about asc / desc to another variable, then use property directly without any if statement